### PR TITLE
(#444) Create getDiseasesForTypeAhead fetcher

### LINKED ISSUE #444: Create getDiseasesForTypeAhead fetcher for v2
# Create getDiseasesForTypeAhead fetcher
> **ESTIMATE** 3

This is a replacement for the middleware action [getDiseasesForSimpleTypeAhead](https://github.com/NCIOCPL/clinical-trials-search-app/blob/develop/src/store/actions.js#L57) that is used to fetch the type ahead condition search on the basic search page.
* Make a fetch command in `src/services/api/clinical-trials-search-api`
   * Call it getDiseasesForTypeAhead or something like that -- the /diseases endpoint is used for MANY fields in this form, so making it purpose built will make life easy.
* Wire up the middleware to use the new fetcher.

> ESTIMATE: TBD

### Acceptance Criteria

```gherkin
Scenario: User searches for main cancer type via autosuggest on basic form
    Given user is navigating to "/about-cancer/treatment/clinical-trials/search"
    When user clicks on "CancerTypeCondition" field
    Then the autocomplete dropdown displays with results starting with "All"
    When user types "le" in "CancerTypeCondition" field
    Then the autocomplete dropdown is displayed with the following results
      | index | value                                   |
      | 0     | Acute Lymphoblastic Leukemia (ALL)      |
      | 1     | Acute Myeloid Leukemia (AML)            |
      | 2     | Chronic Lymphocytic Leukemia (CLL)      |
      | 3     | Chronic Myelogenous Leukemia (CML)      |
      | 4     | Chronic Myelomonocytic Leukemia (CMML)  |
      | 5     | Leiomyosarcoma                          |
      | 6     | Leukemia                                |
      | 7     | Lymphoid Leukemia                       |
      | 8     | Myelodysplastic Syndrome                |
      | 9     | Myeloid Leukemia                        |
```

``` gherkin
 Scenario: User searches for main cancer type that does not exist on basic form
    Given user is navigating to "/about-cancer/treatment/clinical-trials/search"
    When user clicks on "CancerTypeCondition" field
    Then the autocomplete dropdown displays with results starting with "All"
    When user types "asdf" in "CancerTypeCondition" field
    Then autocomplete dropdown is displayed with "No results found" text
```
------
### API Notes
* Request in v1:
``` 
method: GET
endpoint: https://clinicaltrialsapi.cancer.gov/v1/diseases?size=0&current_trial_status[]=Active&current_trial_status[]=Approved&current_trial_status[]=Enrolling+by+Invitation&current_trial_status[]=In+Review&current_trial_status[]=Temporarily+Closed+to+Accrual&current_trial_status[]=Temporarily+Closed+to+Accrual+and+Intervention&type=maintype
```

* Request parameters in v1:
``` 
size: 0
current_trial_status[]: Active
current_trial_status[]: Approved
current_trial_status[]: Enrolling by Invitation
current_trial_status[]: In Review
current_trial_status[]: Temporarily Closed to Accrual
current_trial_status[]: Temporarily Closed to Accrual and Intervention
type: maintype
```

* Response in v1:
``` 
{
  "terms":[
    {
      "name":"Acute Lymphoblastic Leukemia (ALL)",
      "codes":[
        "C3167"
      ],
      "ancestor_ids":[
        "C27134",
        "C3161",
        "C7539"
      ],
      "parent_ids":[
        "C7539"
      ],
      "type":[
        "maintype",
        "subtype"
      ]
    },
    {
      "name":"Acute Myeloid Leukemia (AML)",
      "codes":[
        "C3171"
      ],
      "ancestor_ids":[
        "C27134",
        "C3172",
        "C3161",
        "C9290"
      ],
      "parent_ids":[
        "C3172"
      ],
      "type":[
        "maintype",
        "subtype"
      ]
    },
    {
      "name":"Adrenal Cortex Cancer",
      "codes":[
        "C9325"
      ],
      "ancestor_ids":[
        "C7352"
      ],
      "parent_ids":[
        "C7352"
      ],
      "type":[
        "maintype",
        "subtype"
      ]
    }
  ]
}
 ```

* Request in v2:
``` v2 request```

* Request parameters in v2:
``` v2 params```

* Response in v2:
``` v2 response ```

* Technical notes:
  * Notes here
* Possible mocks:
``` mocks ```

### PR DESCRIPTION
**Pull request by seyilonge-nci**
Closes #444

* Added `getDiseasesForTypeAheadAction` and `getDiseasesForTypeAhead` fetcher
* Added `getDiseasesForTypeAhead` to middleware
* Added unit tests
* Updated `CancerTypeKeyword` to use `getDiseasesForTypeAheadAction`
* Updated `Autocomplete` feature file
* Removed `grade` type from v2 request. 
  
---
* Added mock files for #444

### CODE REVIEW

File: cypress/integration/Components/Autocomplete.feature [hunk 1]
```diff
@@ -90,7 +90,7 @@ Feature: As a user, I want to be able to use autocomplete features to it's full
 
     Scenario: as a user, I should be able to touch screen to select item
         Given the user navigates to "/about-cancer/treatment/clinical-trials/search"
-        When user types "bre" in "Cancer Type/Keyword" field
+        When user types "breast cancer" in "Cancer Type/Keyword" field
```
Comment (kate-mashkina) [inline cypress/integration/Components/Autocomplete.feature:93]: I'm sure you know I am going to ask why this was changed ?)

Comment (arcepaul) [reply cypress/integration/Components/Autocomplete.feature:93]: ..saves me from asking

**Comment by kate-mashkina** [comment]
There are some missing mocks warning:

[Error: ENOENT: no such file or directory, access '/home/runner/work/clinical-trials-search-app/clinical-trials-search-app/support/mock-data/diseases/as_maintype-subtype-stage-grade_10_empty.json'] {
[Error: ENOENT: no such file or directory, access '/home/runner/work/clinical-trials-search-app/clinical-trials-search-app/support/mock-data/diseases/asd_maintype-subtype-stage-grade_10_empty.json'] {
[Error: ENOENT: no such file or directory, access '/home/runner/work/clinical-trials-search-app/clinical-trials-search-app/support/mock-data/diseases/asdf_maintype-subtype-stage-grade_10_empty.json'] {

File: src/store/actionsV2/__tests__/getDiseasesForTypeAheadAction.test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+describe('getDiseasesForTypeAheadAction', () => {
+	it('should match returned object', () => {
+		const searchText = 'Breast Cancer';
+
+		const requestQuery = {
+			name: searchText,
+			current_trial_status: ACTIVE_TRIAL_STATUSES,
+			size: 10,
+			type: ['maintype', 'subtype', 'stage', 'grade'],
```
Comment (kate-mashkina) [inline src/store/actionsV2/__tests__/getDiseasesForTypeAheadAction.test.js:12]: one more thing - is 'grade' even a type? Old action only had ['maintype', 'subtype', 'stage']

File: cypress/integration/Components/Autocomplete.feature (hunk 1 above)
Comment (seyilonge-nci) [reply cypress/integration/Components/Autocomplete.feature:93]: @kate-mashkina @arcepaul ^^^

File: src/store/actionsV2/__tests__/getDiseasesForTypeAheadAction.test.js (hunk 2 above)
Comment (seyilonge-nci) [reply src/store/actionsV2/__tests__/getDiseasesForTypeAheadAction.test.js:12]: I was going by what was in the PP document for v2 request which had `grade` as a type.

Comment (seyilonge-nci) [reply src/store/actionsV2/__tests__/getDiseasesForTypeAheadAction.test.js]: @kate-mashkina, @sarinapadilla confirmed with @bryanpizzillo Type `grade` shouldn't be used for CTS as its intersects with type `stage` and causes confusion.

Removed type `grade` from `v2` request for PR.

**Review by KateMashkinaNIH — APPROVED** [review]

**Review by arcepaul — APPROVED** [review]

**Pull request merged**